src/api/firebase.js

```diff
@@ -15,6 +15,16 @@ export function streamListItems(listId, handleSuccess) {
 	return onSnapshot(listCollectionRef, handleSuccess);
 }
 
+/**
+ * Check existence of list in Firestore associated with user token input.
+ * @param {string} userTokenInput The user's token input requesting to be validated
+ */
+export async function validateToken(userTokenInput) {
+	const listCollectionRef = collection(db, userTokenInput);
+	const listSnapshot = await getDocs(listCollectionRef);
+	return !listSnapshot.empty;
+}
```

Comment:
This function works exactly as intended, but I have an optimization mission, should you choose to accept it.

We know our tokens have to be two things:
1. non-empty, and
2. a specific shape

Can we use this knowledge to check for validity and return early _without_ wasting time and resources on a request to our database?

Reply:
Agh, such a great point EJ! I'll incorporate that now.

Reply:
Actually, before doing so I want to ask a question at today's sync. Going to merge to main for now.

Reply:
Great point, EJ! It would be great to talk this over today.